plutus-tx-plugin/src/Language/PlutusTx/Plugin.hs

```diff
@@ -51,31 +58,22 @@ import qualified Data.Text.Prettyprint.Doc              as PP
 import           GHC.TypeLits
 import           System.IO.Unsafe                       (unsafePerformIO)
 
-{- Note [Constructing the final program]
-Our final type is a simple newtype wrapper. However, constructing *anything* in Core
-is a pain - we have to go off and find the right constructor, ensure we've applied it
-correctly etc. But since it *is* just a wrapper... we can just put in a coercion!
-
-Very nice and easy, but we need to make sure we don't stop being a simple newtype
-without revisiting this.
```

Comment:
This is no longer true?

Reply:
Indeed, now it's a datatype and we actually get the data constructor and construct it properly.